### PR TITLE
Update Windows Installer projects to use ProjectReference

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes:
* Build installer projects in parallel.
* Use `ProjectReference` to ferry bits between installer projects.
* Don't build wixproj to a unified output directory. This was only done to simplify finding file paths to MSI's built by other projects, but ProjectRef solves that.
* Add a VS solution for working on wixproj and the associated C++ custom actions.
   * To make wixproj work in VS, I replaced default globs with listing .wxs and .wxl files in the wixproj file.
* Add a target to copy the installers to the artifacts directory according to the layout described in https://github.com/aspnet/AspNetCore/blob/master/docs/Artifacts.md

Resolves https://github.com/aspnet/AspNetCore-Internal/issues/1571

### CODE REVIEW

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj [hunk 1]
```diff
@@ -56,17 +29,41 @@
...
+  <Target Name="BeforeBuild">
+    <CreateProperty Condition="$(IIS-Setup) != ''" Value="IIS-Setup=$(IIS-Setup);$(DefineConstants)">
+      <Output TaskParameter="Value" PropertyName="DefineConstants" />
+    </CreateProperty>
+  </Target>
+  <Target Name="EnsureImportsExist" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project is trying to import a missing file: {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('..\build\settings.props')" Text="$([System.String]::Format('$(ErrorText)', '..\build\settings.props'))" />
```
Comment (pakrym) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj]: Lets remove conditions from imports all together

**Review by pakrym — APPROVED** [review]
Looks good

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj [hunk 2]
```diff
@@ -56,17 +29,41 @@
...
-  <ItemGroup>
     <Content Include="$(IIS-Setup)include.wxi">
       <Link>include.wxi</Link>
     </Content>
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\CustomAction\aspnetcoreCA.vcxproj">
+      <Name>aspnetcoreCA</Name>
+      <Project>{7c27e72f-54d0-4820-8cfa-5e4be640974b}</Project>
```
Comment (pakrym) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj]: Is this still required?

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj [hunk 3]
```diff
@@ -56,17 +29,41 @@
...
+      <Private>True</Private>
+      <DoNotHarvest>True</DoNotHarvest>
+    </ProjectReference>
+  </ItemGroup>
+
+  <Target Name="BeforeBuild">
+    <CreateProperty Condition="$(IIS-Setup) != ''" Value="IIS-Setup=$(IIS-Setup);$(DefineConstants)">
```
Comment (pakrym) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj]: Is `IIS-Setup` ever set? Let's remove this target if not.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj (hunk 2 above)
Comment (natemcmaster) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMIISExpressV1/AncmIISExpressV1.wixproj]: Not required for command line builds, only for opening this project in Visual Studio.

**Review by jkotalik — APPROVED** [review]

**Comment by jkotalik** [comment]
Can this be backported to 2.1/2.2?

**Comment by natemcmaster** [comment]
Yeah, that's a good idea. It's definitely lower priority for me thought, so I'm going to open this to track this work: https://github.com/aspnet/AspNetCore-Internal/issues/1581

**Pull request merged**